coder-sdk/error.go

```diff
@@ -17,31 +17,52 @@ var ErrPermissions = xerrors.New("insufficient permissions")
 // ErrAuthentication describes the error case in which the requester has invalid authentication.
 var ErrAuthentication = xerrors.New("invalid authentication")
 
-// apiError is the expected payload format for our errors.
-type apiError struct {
-	Err apiErrorMsg `json:"error"`
+// APIError is the expected payload format for our errors.
+type APIError struct {
+	Err APIErrorMsg `json:"error"`
 }
 
-// apiErrorMsg contains the rich error information returned by API errors.
-type apiErrorMsg struct {
-	Msg string `json:"msg"`
+// APIErrorMsg contains the rich error information returned by API errors.
+type APIErrorMsg struct {
+	Msg     string          `json:"msg"`
+	Code    string          `json:"code"`
+	Details json.RawMessage `json:"details"`
 }
 
 // HTTPError represents an error from the Coder API.
 type HTTPError struct {
 	*http.Response
+	cached    *APIError
+	cachedErr error
```

Comment:
We decode the `re.Response.Body` on `.Error()`. I didn't feel like trying to fix that behavior, but I need to cache the decoded payload as .Error() can be called more than once, and then the second one fails. So I split `.Error()` in to `.Payload` so I can extract the payload and not change any current behavior.

Reply:
oh nice catch, I also think we had this buggy code duplicated in the monorepo. merged a patch there last week.

Reply:
It still feels a little odd, but I didn't want to refactor how errors worked as I only touched small areas of the cli so far.
So this works for now.